src/main/java/com/vmware/vim25/mo/VirtualMachineSnapshot.java

```diff
@@ -36,70 +36,73 @@
-	{
-	  ManagedObjectReference mor = getVimService().exportSnapshot(this.getMOR());
-	  return new HttpNfcLease(this.getServerConnection(), mor);
-	}
-	
-  //SDK4.1 signature for back compatibility
-	public Task removeSnapshot_Task(boolean removeChildren) throws TaskInProgress, RuntimeFault, RemoteException 
-	{	
-	  return removeSnapshot_Task(removeChildren, null);
-	}
-
-  //SDK5.0 signature
-	public Task removeSnapshot_Task(boolean removeChildren, Boolean consolidate) throws TaskInProgress, RuntimeFault, RemoteException 
-	{ 
-	  return new Task( getServerConnection(),
-	      getVimService().removeSnapshot_Task(getMOR(), removeChildren, consolidate));
-	}
-	
-	public void renameSnapshot(String name, String description) throws InvalidName, RuntimeFault, RemoteException 
-	{
-		getVimService().renameSnapshot(getMOR(), name, description);
-	}
-	
-	//SDK2.5 signature for back compatibility
-	public Task revertToSnapshot_Task(HostSystem host) throws VmConfigFault, TaskInProgress, FileFault, InvalidState, InsufficientResourcesFault, RuntimeFault, RemoteException 
-	{
-		return revertToSnapshot_Task(host, null);
-	}
-	
-	//SDK4.0 signature
-	public Task revertToSnapshot_Task(HostSystem host, Boolean suppressPowerOn) throws VmConfigFault, TaskInProgress, FileFault, InvalidState, InsufficientResourcesFault, RuntimeFault, RemoteException 
-	{
-		return new Task(getServerConnection(),
-				getVimService().revertToSnapshot_Task(getMOR(), host==null? null: host.getMOR(), suppressPowerOn));
-	}
 
+    //SDK5.0 signature
+    public Task removeSnapshot_Task(boolean removeChildren, Boolean consolidate) throws TaskInProgress, RuntimeFault, RemoteException {
+        return new Task(getServerConnection(),
+            getVimService().removeSnapshot_Task(getMOR(), removeChildren, consolidate));
+    }
+
+    public void renameSnapshot(String name, String description) throws InvalidName, RuntimeFault, RemoteException {
+        getVimService().renameSnapshot(getMOR(), name, description);
+    }
+
+    //SDK2.5 signature for back compatibility
+    public Task revertToSnapshot_Task(HostSystem host) throws VmConfigFault, TaskInProgress, FileFault, InvalidState, InsufficientResourcesFault, RuntimeFault, RemoteException {
+        return revertToSnapshot_Task(host, null);
+    }
+
+    //SDK4.0 signature
+    public Task revertToSnapshot_Task(HostSystem host, Boolean suppressPowerOn) throws VmConfigFault, TaskInProgress, FileFault, InvalidState, InsufficientResourcesFault, RuntimeFault, RemoteException {
+        return new Task(getServerConnection(),
+            getVimService().revertToSnapshot_Task(getMOR(), host == null ? null : host.getMOR(), suppressPowerOn));
+    }
+
+    /**
+     * Get the virtual machine for which the snapshot was taken.
+     *
+     * @since 6.0
+     * @return {@link com.vmware.vim25.mo.VirtualMachine VirtualMachine}
+     */
+    public VirtualMachine getVm() {
+        return (VirtualMachine) getCurrentProperty("vm");

```

Comment:
After some review I think this may need to be converted from a MOR to a MO using the MorUtil. Review further before merging.